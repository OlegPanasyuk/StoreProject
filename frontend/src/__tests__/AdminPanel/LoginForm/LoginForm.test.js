import React from 'react';
import LoginFormConnected, { LoginForm } from '../../../AdminPanel/LoginForm/LoginForm';
import { shallow, configure, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

configure({ adapter: new Adapter() });

function setupComponent(props = {
    match: {
        path: ''
    },
    user: {

    },
    userAuthorizedSuccess: () => { }
}) {
    let component = shallow(<LoginForm {...props} />);
    return { component, props };
}

describe('LoginForm --- Snapshot', () => {
    it('Component must be rendered', () => {
        let { component } = setupComponent();
        expect(LoginForm.propTypes.match).toBe(PropTypes.object);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});


describe('LoginForm ', () => {
    let initialState = {
        adminPanel_User: {

        }
    };

    const mockStore = configureStore();
    let store, wrapper;

    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = mount(<Provider store={store}><LoginFormConnected /></Provider>);
    });
    it('+++ render the connected(SMART) component', () => {
        expect(wrapper.find(LoginFormConnected).length).toEqual(1);
    });
});

