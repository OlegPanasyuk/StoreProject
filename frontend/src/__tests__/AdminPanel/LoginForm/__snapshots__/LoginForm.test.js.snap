// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm --- Snapshot Component must be rendered 1`] = `
<Fragment>
  <Bootstrap(Modal)
    centered={true}
    keyboard={true}
    onEntering={[Function]}
    onExit={[Function]}
    onHide={[Function]}
    show={true}
    size="sm"
  >
    <ModalHeader
      closeButton={false}
      closeLabel="Close"
    >
      <h4>
        LoginForm
      </h4>
    </ModalHeader>
    <Form
      as="form"
      inline={false}
    >
      <ModalBody>
        <FormGroup
          as="div"
        >
          <FormControl
            as="input"
            placeholder="Enter email"
            type="email"
          />
        </FormGroup>
        <FormGroup
          as="div"
        >
          <FormControl
            as="input"
            placeholder="Password"
            type="password"
          />
        </FormGroup>
      </ModalBody>
      <ModalFooter>
        <Button
          active={false}
          disabled={false}
          id="buttonToLogin"
          onClick={[Function]}
          type="button"
          variant="primary"
        >
          Login
        </Button>
        <Button
          active={false}
          disabled={false}
          onClick={[Function]}
          type="button"
          variant="light"
        >
          Close
        </Button>
      </ModalFooter>
    </Form>
  </Bootstrap(Modal)>
  <Overlay
    placement="left"
    rootClose={false}
    show={false}
    target={null}
    transition={[Function]}
  >
    <Component />
  </Overlay>
</Fragment>
`;
